libqtile/widget/mpris2widget.py

```diff
@@ -68,25 +76,50 @@ def __init__(self, **config):
         self.is_playing = False
         self.scroll_timer = None
         self.scroll_counter = None
-        self.dbus_loop = None
-
-    def _configure(self, qtile, bar):
-        base._TextBox._configure(self, qtile, bar)
-
-        # we don't need to reconnect all the dbus stuff if we already
-        # connected it.
-        if self.dbus_loop is not None:
+        self.count = 0
+        self.connect_count = 0
+
+        self.match_args = {
+            "type": "signal",
+            "sender": self.objname,
+            "member": "PropertiesChanged",
+            "path": "/org/mpris/MediaPlayer2",
+            "interface": "org.freedesktop.DBus.Properties"
+        }
+
+    def _eventloop_started(self):
+        self._connect_dbus()
+
+    def _connect_dbus(self):
+        if self.connect_count < self.MAX_RETRIES:
+            self.connect_count += 1
+            asyncio.create_task(self._setup())
+        else:
+            logger.warning("Exceeded max retries. "
+                           "Could not subscribe to dbus signals.")
+
+    async def _setup(self):
+        self.bus = await MessageBus().connect()
+        rule = ",".join("{}='{}'".format(k, v)
+                        for k, v in self.match_args.items())
+        subscribe = await self.bus.call(
+            Message(destination='org.freedesktop.DBus',
+                    interface='org.freedesktop.DBus',
+                    path='/org/freedesktop/DBus',
+                    member='AddMatch',
+                    signature='s',
+                    body=[rule]))
```

Comment:
Is there anything higher level than this in dbus-next? Something closer to the original `add_signal_receiver` method?

Reply:
Excellent question and I asked something very similar on the dbus_next repo: https://github.com/altdesktop/python-dbus-next/issues/75 and https://github.com/altdesktop/python-dbus-next/issues/76

The second link shows that there's no higher level method currently but this might change in the future.

Reply:
Then we should probably add a small function in `libqtile.utils` that does just this, that we could propose upstream later. What do you think?

Reply:
That would be a neater way of doing it, especially as it's code that could be used by other widgets. 

I'll knock something up and add it to this PR.

Reply:
I've left these as separate commits for now so you can see the changes I've made. Can squash them later.

Reply:
Thanks!